Bài Tập của Đức

vanduc2514 opened the pull request:
Em gửi chị Source code, chị xem và cho ý kiến nhé. Em cám ơn chị!

vanduc2514 committed:
Source code Geometric Object

Em gửi chị Source code, chị xem và cho ý kiến nhé. Em cám ơn chị!

vanduc2514 committed:
Bổ sung Interface, cấu trúc lại hàm Main

Em bổ sung thêm Interface và cấu trúc lại code cho hợp lý. Chị kiểm tra cả commit này nhé. Em cám ơn chị!

vanduc2514 committed:
Source code Student Management

Chị xem và cho ý kiến nhé.

vanduc2514 committed:
Refactor source code Student Management

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+    }
+
+    public Student searchStudentByID(List<Student> list) {
+        System.out.print("Enter Student's ID: ");
+        String id = scanner.next();
```
2 dòng này đưa vào trong hàm main

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+        }
+    }
+
+    public Student searchStudentByID(List<Student> list) {
```
truyền tham thêm tham số id cho hàm

vanduc2514 replied:
uh nhỉ, thế thì code xuôi hơn

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+    }
+
+    public Student searchStudentByName(List<Student> list) {
+        System.out.print("Enter Student's name: ");
+        String name = scanner.nextLine();
```
Để 2 dòng này trong main

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+        return null;
+    }
+
+    public Student searchStudentByName(List<Student> list) {
```
truyền thêm tham số String name

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+        return null;
+    }
+
+    public void updateStudentByID(List<Student> list) {
```
truyền thêm tham số int id

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+        }
+    }
+
+    public void updateStudentByName(List<Student> list) {
```
thêm tham số của hàm là String name

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+        }
+    }
+
+    public void removeStudentByID(List<Student> list) {
```
thêm tham số đầu vào là id

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+        }
+    }
+
+    public void removeStudentByName(List<Student> list) {
```
thêm tham số đầu vào là name

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+    }
+
+    public void addStudentByIndex(List<Student> list) {
+        System.out.print("Enter position you want to add: ");
+        int index = scanner.nextInt();
```
đưa 2 dòng này vào trong main

quannguyen206 commented on src/controller/StudentManagement.java:
```diff
@@ -1,17 +1,132 @@
+        }
+    }
+
+    public void addStudentByIndex(List<Student> list) {
```
thêm tham số đầu vào là index

quannguyen206 commented on test/Main.java:
```diff
@@ -1,94 +1,104 @@
-        System.out.println("0. Exit Program!");
-        System.out.print("Enter choice: ");
+        while (!isExit) {
+            System.out.println("---------------------------------");
+            System.out.println("1. Add Student");
+            System.out.println("2. Search Student's information by id");
+            System.out.println("3. Search Student's information by name");
+            System.out.println("4. Edit Student's information by id");
+            System.out.println("5. Edit Student's information by name");
+            System.out.println("6. Delete Student by id");
+            System.out.println("7. Delete Student by name");
+            System.out.println("8. Display List Student");
+            System.out.println("9. Display Female Student");
+            System.out.println("10. Display Student has Scholarship");
+            System.out.println("11. Add Student by position in List");
+            System.out.println("0. Exit Program!");
```
tách ra thành 1 hàm menu

quannguyen206 commented on test/Main.java:
```diff
@@ -97,6 +107,7 @@ private static boolean isValidChoice(int choice) {
                 return true;
             }
         }
+        System.out.println("Invalid Choice! Please choose again!");
```
không để in ra màn hình ở đây đưa vào trong main

vanduc2514 replied:
là sao nhỉ

vanduc2514 committed:
Bài tập Binary Search Tree và Binary Search

vanduc2514 committed:
Source code lớp cắt chuỗi theo thứ tự alphabet

vanduc2514 committed:
Các thuật toán sắp xếp và minh hoạ

vanduc2514 committed:
Refactor Source Binary Search tree, thêm phương thức, test case

1. Thêm phương thức tìm kiếm đệ quy
2. Thêm phương thức kiểm tra size sau khi xoá
3. Sửa code chèn đệ quy để tăng size sau khi chèn
4. Sửa code Tìm kiếm nhị phân đệ quy

vanduc2514 committed:
Refactor source code Student management	

1. Tái cấu trúc
2. Thêm Comparable và Comparator
3. Sửa lại menu

vanduc2514 committed:
Source Code Xử lý File văn bản

vanduc2514 committed:
Source Code các bài tập trong phần IOStream

quannguyen206 reviewed (commented):
đoạn code này của a tốt đã tách hàm, tách chức năng đầy đủ

vanduc2514 commented:
> đoạn code này của a tốt đã tách hàm, tách chức năng đầy đủ

Anh đang update thêm mấy tính năng nữa, lát em xem cho anh nha, còn mỗi cái sắp xếp là xong

vanduc2514 committed:
Source Code phần Đa luồng và đồng bộ

vanduc2514 committed:
Source Code phần đa luồng và đồng bộ

vanduc2514 committed:
Source Code phần Regex trong Java
